### ISSUE TITLE
Notify when SSL/TLS certificate changes

### ISSUE THREAD

**Issue by MrEddX** [louislam/uptime-kuma#4143]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Notification, New Monitor

### 🔖 Feature description

Option to send a message in case the current certificate is changed/renewed. Maybe somewhere in the "Аdvanced" settings, under the "Certificate Expiry Notification" or "Ignore TLS/SSL error for HTTPS websites" check box in the monitor settings.

![image](https://github.com/louislam/uptime-kuma/assets/66828538/1f5df853-3426-4187-8622-578b3cd2ed94)

### ✔️ Solution

I have no idea if it's possible and how laborious it would be.

### 📝 Additional Context

Maybe it would help to more quickly reveal situations like: [Mitigating the Hetzner/Linode XMPP.ru MitM interception incident](https://news.ycombinator.com/item?id=37961166)

**Comment by ChatrixOne** [comment]
It would be quite useful! I vote for this.

**Comment by Goozoon** [comment]
Yes, would be nice to get an info about new Let's Ecrypt Cert. In not so long time those will have even shorter lifespan

**Comment by Aleixenandros** [comment]
It would be great to have an option (for example, in the Advanced section of the HTTPS monitor) to enable notifications when the TLS certificate changes or is renewed, even if it is not about to expire.

Ideally, the notification could include:

- Old fingerprint (SHA256)
- New fingerprint (SHA256)
- Possibly other details such as Subject, Issuer or validity dates
This allows the detection of automatic renewals (e.g., Let's Encrypt) to confirm that everything is working correctly, or unexpected changes that could indicate a security problem (MITM, etc.), similar to what Zabbix does when it warns of a fingerprint change.

**Comment by CommanderStorm** [comment]
I think we can offload this one to certificate transparency instead of having to add more surounding this feature.

-> https://de.wikipedia.org/wiki/Certificate_Transparency

In my mind, this is fairly a niche attack.
If someone else has enough server access to your site to issue valid certificates for it, I think you have bigger problems and the noise of such a system would generally not be desirable imo.
I am going to mark this as not planned unless someone with security background argues hard against this.
If you are arguing against this, consider joining forces:
- https://github.com/louislam/uptime-kuma/issues/6673

**Comment by Aleixenandros** [comment]
Thanks for the feedback! I think there might be a slight misunderstanding about what I'm proposing here.

This isn't really about catching MITM attacks or security breaches - it's more about **day-to-day operational monitoring**, especially for those of us managing auto-renewing certificates like Let's Encrypt.

**Here's why I think it would be useful:**

1. **Peace of mind that auto-renewal is working**: When you have Let's Encrypt (or similar) handling renewals automatically, it's really nice to get a quick notification saying "Hey, your cert just renewed successfully!" It's just good operational feedback.

2. **Keep it simple**: I'm not asking for anything fancy - just a notification like "Certificate fingerprint changed for [domain]. New certificate valid for X days" would be perfect.

3. **Catch issues early**: Right now, we only get alerts when expiration is close. If auto-renewal silently fails, you're already in emergency mode by the time you find out. A renewal notification means you know everything's working as it should.

4. **Pretty standard feature**: A lot of monitoring tools (like Zabbix) include this because it's genuinely helpful for operations, not just security paranoia.

**About Certificate Transparency**: CT logs are great for their purpose, but they're a separate thing you'd need to monitor with different tools. Part of what makes Uptime Kuma awesome is having everything in one place with your preferred notification channels already set up.

**Implementation suggestion**:

Looking at the codebase, this could be implemented fairly simply as an optional feature:

1. **Storage**: The monitor already captures `tlsInfo` (including fingerprint) - we'd just need to store the previous fingerprint in the monitor's state (maybe in the `monitor` table as a new field like `last_tls_fingerprint`).

2. **Detection logic**: In the monitor check cycle (wherever `tlsInfo` is currently updated), compare the new fingerprint with the stored one. If different, trigger a notification event.

3. **UI toggle**: Add a checkbox in the Advanced section of HTTP(S) monitors: "Notify on certificate change" (disabled by default to avoid noise for those who don't need it).

4. **Notification**: Use the existing notification system with a new event type (like `certChanged`) that includes:
   - Monitor name
   - Old fingerprint (SHA256)
   - New fingerprint (SHA256)  
   - New expiry date
   - Days until expiration

This would integrate naturally with the existing notification infrastructure and wouldn't add much complexity. For those who don't need it, it's just an unchecked box they'll never notice. But for those managing multiple domains with auto-renewal, it would be super helpful!

What do you think?

**Comment by CommanderStorm** [comment]
I don't think this is a usefull signal.
"Hey, your cert just renewed successfully" is very spammy in the day and age of 7-day certificates.

**Comment by Aleixenandros** [comment]
> I don't think this is a usefull signal. "Hey, your cert just renewed successfully" is very spammy in the day and age of 7-day certificates.

the proposal is to write a check on demand.

**Comment by shalafi99** [comment]
For now, I have been using the workaround below for somewhat close functionality.
All my certs are from Let's encrypt and none are from the "short-lived" kind, hence just those 2 entries suit my environment.
I know this is a "hack" but wanted to share nevertheless.

<img width="507" height="261" alt="Image" src="https://github.com/user-attachments/assets/0e161f65-4551-4d5c-9836-20936b868222" />
